src/shuffile.h

```diff
@@ -22,12 +21,41 @@ extern "C" {
 #define SHUFFILE_SUCCESS (0)
 #define SHUFFILE_FAILURE (1)
 
+#define SHUFFILE_KEY_CONFIG_MPI_BUF_SIZE "MPI_BUF_SIZE"
+#define SHUFFILE_KEY_CONFIG_DEBUG "DEBUG"
+
 /** initialize library */
 int shuffile_init();
 
 /** shutdown library */
 int shuffile_finalize();
 
+/* needs to be above doxygen comment to get association right */
+typedef struct kvtree_struct kvtree;
+/**
+ * Get/set shuffile configuration values.
+ *
+ * The following configuration options can be set (type in parenthesis):
+ *   * "DEBUG" (int) - if non-zero, output debug information from inside
+ *     shuffile.
+ *   * "MPI_BUF_SIZE" (byte count [IN], int [OUT]) - MPI buffer size to chunk
+ *     file transfer. Must not exceed INT_MAX.
+ *   .
+ * Symbolic names SHUFFILE_KEY_CONFIG_FOO are defined in shuffile.h and should
+ * be used instead of the strings whenever possible to guard against typos in
+ * strings.
+ *
+ * \result If config != NULL, then return config on success.  If config=NULL
+ *         (you're querying the config) then return a new kvtree on success.
+ *         NULL on any failures.
+ * \param config The new configuration. If config=NULL, then return a kvtree
+ *                with all the configuration values.
```

Comment:
Let's mention the caller is responsible to delete returned kvtree.

Reply:
Added in 4cbcf0e